### ISSUE TITLE
default impacting dependsOn/exclusive

### ISSUE THREAD

**Issue by mshanemc** [oclif/oclif#211]
Not sure if this is a bug or a feature, it's more like a bug in between existing features.

### Summary

if I set a default on a flag, the parser is acting like it's been set by the user (interfering with exclusive/dependsOn properties.

### Steps To Reproduce:

```
protected static flagsConfig = {
    object: flags.string({char: 'o', description: 'what object to check perms on', required: true}),
    field: flags.string({char: 'f', description: 'what field to check'}),
    profiles: flags.boolean({description: 'return names/ids of profiles'}),
    permsets: flags.boolean({description: 'return names/ids of permission sets'}),
    users: flags.boolean({description: 'return names/ids of users with those profiles and/or permission sets', exclusive: ['permsets', 'profiles']}),
    fieldlevel: flags.string({description: 'what level of perms are you looking for', default: 'Read', options: ['Read', 'Edit'], dependsOn: ['field']}),
    objectlevel: flags.string({description: 'what level of perms are you looking for', default: 'Read', options: ['Read', 'Edit', 'Create', 'Delete', 'ViewAll', 'ModifyAll'], exclusive: ['field']})
  };
```

In this example, if I **don't** specify a value for objectlevel, because it has a default, I'm getting an error thrown

`$ sfdx shane:permset:check -o Project__c -u platformers --field Due_Date__c --fieldlevel Edit --profiles --permsets`
`ERROR running PermCheck:  --field= cannot also be provided when using --objectlevel=`

### Expected result

don't enforce the exclusive unless I'm actually providing a value

### Actual result

exclusive and dependsOn are enforced

**SFDX CLI Version**:
sfdx-cli/6.53.0 (darwin-x64) node-v11.7.0

**OS and version**:
macOS/Mojave 10.14.3

**Comment by elbandito** [comment]
@mshanemc thanks for the info.  This is definitely a tricky one, because we don't have a straight forward way of knowing if a flag's value was defaulted.  However, we are looking into it.

In the meantime, one way around this is to remove the `default` attribute from `objectlevel` and then do something along the lines of:
```
const objectlevel = flags.objectlevel || 'Read'
```
Travis

**Comment by elbandito** [comment]
@mshanemc FYI https://github.com/oclif/parser/pull/45

I don't know the release cadence for `sfdx`, but once they update/release it should be fixed.

**Comment by jdxcode** [comment]
the parser is part of the command—not the CLI, so it doesn't matter what version sfdx is on, just what version the plugin is using

**Comment by mshanemc** [comment]
@jdxcode I'm using Flags from sfdxCommand...wouldn't I be waiting on that to release a version?

**Comment by jdxcode** [comment]
depends on whether sfdx is locking the dependencies down to a specific version of `@oclif/command` or not. If it's `^` then you should be able to update the transitive dependency with `yarn upgrade`

**Comment by 0xdevalias** [comment]
@mdonnalley It would be nice to include some context/description on why the issue is being closed (if it was completed, then when/where it was completed; if it's 'not planned' then more of an explanation of why it's not going to happen). This current process of 'close a huge pile of issues' with no context is generally considered pretty 'hostile'/'disrespectful' (for lack of a better term) to open source/collaboration/etc.

**Comment by vassudanagunta** [comment]
Yes, at least give issues that are closed so summarily a label. oclif as [a number of labels already defined](https://github.com/oclif/oclif/labels), including `wontfix` and `duplicate`.

I am not yet a oclif user, but I browse the issues to help me decide the in-house culture and process of the maintainer(s).

**Comment by vassudanagunta** [comment]
@mdonnalley It looks like it was fixed in a commit referenced in an earlier comment: https://github.com/oclif/parser/pull/45.  I think if you read the comments you would have picked up on that.

That said, @mdonnalley, I recommend closing issues with an explicit comment to avoid the appearance of closing things summarily.
